tests/unit/search/test_search_query.py

```diff
@@ -115,6 +115,41 @@ def test_author_bai():
     assert expected == result
 
 
+def test_author_bai():
+    query = IQ('find a r.j.hill.1')
+
+    expected = {
+        "bool": {
+            "should": [
+              {
+                "bool": {
+                  "should": [
+                    {
+                      "term": {
+                        "authors.name_variations": "r.j.hill.1"
+                      }
+                    },
+                    {
+                      "term": {
+                        "authors.inspire_bai": "r.j.hill.1"
+                      }
+                    }
+                  ]
+                }
+              },
+              {
+                "match": {
+                  "authors.full_name": "r.j.hill.1"
+                }
+              }
+            ]
+        }
+    }
```

Comment:
@jmartinm If this is what we were discussing the other day I think it should be like: 

```
    expected = {
        "bool": {
            "must": [
              {
                "bool": {
                  "should": [
                    {
                      "term": {
                        "authors.name_variations": "r.j.hill.1"
                      }
                    },
                    {
                      "term": {
                        "authors.inspire_bai": "r.j.hill.1"
                      }
                    }
                  ]
                }
              }],
              "should": [
              {
                "match": {
                  "authors.full_name": "r.j.hill.1"
                }
              }]
            
        }
    }
```
Since what we want is that inspire_bai `must` be in `either (should)` "authors.name_variations" or "authors.inspire_bai" and it `would also be good (should)` for it to be matching the "authors.full_name" .